frontend/packages/topology/src/components/quick-search/QuickSearchDetails.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { history, removeQueryArgument } from '@console/internal/components/utils';
+import { CatalogItem } from '@console/plugin-sdk';
+import {
+  Button,
+  ButtonVariant,
+  Text,
+  TextContent,
+  TextVariants,
+  Title,
+} from '@patternfly/react-core';
+import './QuickSearchDetails.scss';
+
+interface QuickSearchDetailsProps {
+  selectedItem: CatalogItem;
+}
+
+const QuickSearchDetails: React.FC<QuickSearchDetailsProps> = ({ selectedItem }) => {
+  const { t } = useTranslation();
+
+  return (
+    <div className="odc-quick-search-details">
+      <Title headingLevel="h4">{selectedItem?.name}</Title>
+      {selectedItem?.provider && (
+        <span className="odc-quick-search-details__provider">
+          {t('topology~Provided by')} {selectedItem.provider}
+        </span>
+      )}
+      <Button
+        variant={ButtonVariant.primary}
+        className="odc-quick-search-details__form-button"
+        onClick={() => {
+          removeQueryArgument('catalogSearch');
+          history.push(selectedItem?.cta?.href);
+        }}
+      >
+        {selectedItem?.cta?.label}
+      </Button>
+      <TextContent>
+        <Text component={TextVariants.p} className="odc-quick-search-details__description">
+          {selectedItem?.description}
```

Comment:
I see this a lot. The props say that `selectedItem` is required yet there's optional chaining here.
```suggestion
          {selectedItem.description}
```

Reply:
This was not resolved @divyanshiGupta

Reply:
@andrewballantyne this is resolved as I have made selectedItem optional in the props. But now I think a better solution would be to add a check for it in the parent itself so that it always has a selectedItem.

Reply:
Done.